### PR TITLE
feat: add CDK synthesis validation tests

### PR DESCRIPTION
**Pull request by notgitika**
## Summary
- Adds 19 CDK synthesis smoke tests that validate `agentcore.json` spec shapes can be synthesized into valid CloudFormation templates via `AgentCoreStack` → `AgentCoreApplication` L3 construct
- Covers agents (CodeZip, Container, MCP/A2A protocols), memories, credentials, evaluators, online eval configs, policy engines, and edge cases
- Adds `@aws/agentcore-cdk` resolve alias in `vitest.config.ts` to handle CJS-only package exports in Vite

## Test plan
- [x] All 19 new tests pass (`npx vitest run src/assets/__tests__/cdk-synth-validation.test.ts`)
- [x] Existing unit tests unaffected (pre-existing failures only)
- [x] Lint, prettier, typecheck, secretlint all pass (verified by pre-commit hook)

### CODE REVIEW

**Review by Hweinstock — COMMENTED** [review]
Thanks for adding the tests! Wondering if its possible to remove some the casting to get some stronger compile time checks?

File: src/assets/__tests__/cdk-synth-validation.test.ts [hunk 1]
```diff
@@ -0,0 +1,480 @@
...
+
+function synthStack(
+  spec: AgentCoreProjectSpec,
+  mcpSpec?: unknown,
+  credentials?: Record<string, { credentialProviderArn: string; clientSecretArn?: string }>
+): Template {
+  const app = new cdk.App();
+  const stack = new AgentCoreStack(app, `TestStack${Date.now()}`, {
+    spec,
+    mcpSpec: mcpSpec as never,
```
[the remark this answers is not in the document]
Comment (notgitika) [reply src/assets/__tests__/cdk-synth-validation.test.ts]: removed

File: src/assets/__tests__/cdk-synth-validation.test.ts [hunk 2]
```diff
@@ -0,0 +1,480 @@
...
+    name: 'testproject',
+    version: 1,
+    agents: [],
+    memories: [],
+    credentials: [],
+    evaluators: [],
+    onlineEvalConfigs: [],
+    policyEngines: [],
+    ...overrides,
+  } as AgentCoreProjectSpec;
```
[the remark this answers is not in the document]
Comment (notgitika) [reply src/assets/__tests__/cdk-synth-validation.test.ts]: yes, removed

File: src/assets/__tests__/cdk-synth-validation.test.ts [hunk 3]
```diff
@@ -0,0 +1,480 @@
...
+  for (const dir of agentNames) {
+    mkdirSync(join(tmpDir, 'agents', dir), { recursive: true });
+    writeFileSync(join(tmpDir, 'agents', dir, 'main.py'), '# placeholder');
+    writeFileSync(join(tmpDir, 'agents', dir, 'pyproject.toml'), minimalPyproject);
+    writeFileSync(join(tmpDir, 'agents', dir, 'Dockerfile'), 'FROM python:3.12-slim\n');
+  }
+  // Tell the CDK L3 construct where the project root is so findConfigRoot() succeeds
+  setSessionProjectRoot(tmpDir);
+  originalCwd = process.cwd();
+  process.chdir(tmpDir);
```
[the remark this answers is not in the document]
Comment (notgitika) [reply src/assets/__tests__/cdk-synth-validation.test.ts]: Removed `setSessionProjectRoot()` is sufficient on its own
